File: Calculator.java
//This time, we'll use a switch-case condition instead of an if-else statement as it is more efficient than a block of nested-ifs
import java.util.Scanner;

public class Calculator {
    public static void main(String args[]){
        System.out.println("CALCULATOR");
        double num1;
        double num2;
        String operation;
        Scanner input = new Scanner(System.in);
        System.out.println("Enter 1st Number:");
        num1 = input.nextDouble();
        System.out.println("Enter 2nd Number:");
        num2 = input.nextDouble();
        System.out.println("Enter An Operation From Addition (+), Subtraction (-), Multiplication (*) or Division (/):");
        operation = input.next();
        switch (operation){
            case "+":
                double sum = num1+num2;
                System.out.println("The sum is: "+sum);
                break;
            case "-":
                double difference = num1-num2;
                System.out.println("The difference is: "+difference);
                break;
            case "*":
                double product = num1*num2;
                System.out.println("The product is: "+product);
                break;
            case "/":
                double quotient = num1/num2;
                System.out.println("The quotient is: "+quotient);
                break;
            default:
                System.out.println("That's an invalid input!");
        }
    }
}

File: calculator.cpp
#include <iostream>
using namespace std;

int main(){
    double num1, num2;
    char operation;
    cout<<"CALCULATOR:\n";
    cout<<"Enter the 1st number below:\n";
    cin>>num1;
    cout<<"Enter the 2nd number below:\n";
    cin>>num2;
    cout<<"Instructions for selecting the operation:\n Addition : + \n Subtraction : - \n Multiplication : * \n Division : /:\n";
    cout<<"Enter any 1 out of the operations given above:\n";
    cin>>operation;

    switch (operation){
        case '+':
        cout<<"The sum is: "<<num1+num2<<endl;
        break;
        case '-':
        cout<<"The difference is: "<<num1-num2<<endl;
        break;
        case '*':
        cout<<"The product is: "<<num1*num2<<endl;
        break;
        case '/':
        cout<<"The quotient is: "<<num1/num2<<endl;
        break;
        default:
        cout<<"That's an invalid input!";
        
    }
}